library/Booster/JsonRpc.hs

```diff
@@ -99,7 +100,11 @@ respond stateVar =
                 Left patternError -> do
                     Log.logDebug $ "Error internalising cterm" <> Text.pack (show patternError)
                     pure $ Left $ RpcError.backendError RpcError.CouldNotVerifyPattern patternError
-                Right (pat, substitution) -> do
+                Right (pat, substitution, unsupported) -> do
+                    unless (null unsupported) $
+                        Log.logWarnNS
+                            "booster"
+                            "Execute: ignoring unsupported predicate parts"
```

Comment:
any chance we could maybe make this a custom log level and print the offending term? I suspect this might be a source of "why isn't my state being pruned" issues, because we fail to internalise some constraint and thus don't send it to the solver...

Reply:
I introduced a new `ErrorDetails` level to print more things whenever a `logWarn` or `logError` occurs.
For the case of `simplify` in the `unsupported-predicate` test it prints:
```
[Info#booster] Simplifying predicates
[Warn#booster] Simplify: ignoring unsupported predicates in input
[ErrorDetails#booster] #Not{SortGeneratedTopCell()}(#And{SortGeneratedTopCell()}(#Ceil{SortInt(),SortGeneratedTopCell()}(#DV{SortInt()}(0)),#Equals{SortInt(),SortGeneratedTopCell()}(VarX:SortInt(),#DV{SortInt()}(0))))
```
1. We should probably upstream the `prettyPattern` function.
2. I wasn't sure whether we should print parseable indented kore format or this slightly simpler one in single lines.

Reply:
When eyeballing the logs, it is unlikely that we would benefit from more information than a couple of top-level constructors. I think a single line is better, provided that we have a way to pretty-print it for deeper inspection.